google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/apis/ServiceAccountKeyUltimateDisplayDialog.java

```diff
@@ -0,0 +1,303 @@
+      String executorId, RunnerAndConfigurationSettings configuration) {
+    ProgramRunner runner = ProgramRunnerUtil.getRunner(executorId, configuration);
+    if (runner == null) {
+      LOG.error(
+          "Error updating "
+              + configuration.getName()
+              + " with Google Cloud Library environment variables");
+      return;
+    }
+
+    RunnerSpecificLocalConfigurationBit configurationSettings =
+        (RunnerSpecificLocalConfigurationBit) configuration.getConfigurationSettings(runner);
+    if (configurationSettings == null) {
+      LOG.error(
+          "Error updating "
+              + configuration.getName()
+              + " with Google Cloud Library environment variables");
+      return;
+    }
+
+    Set<EnvironmentVariable> serviceAccountEnvironmentVariables =
+        commonPanel.getEnvironmentVariables();
+    List<EnvironmentVariable> configurationSettingsEnvVariables =
+        configurationSettings.getEnvVariables();
+
+    serviceAccountEnvironmentVariables.forEach(
+        serviceAccountEnvVar -> {
+          OptionalInt indexOpt =
+              IntStream.range(0, configurationSettingsEnvVariables.size())
+                  .filter(
+                      index ->
+                          serviceAccountEnvVar
+                              .getName()
+                              .equals(configurationSettingsEnvVariables.get(index).getName()))
+                  .findFirst();
+          if (indexOpt.isPresent()) {
+            configurationSettingsEnvVariables.set(indexOpt.getAsInt(), serviceAccountEnvVar);
+          } else {
+            configurationSettingsEnvVariables.add(serviceAccountEnvVar);
+          }
+        });
+    configurationSettings.setEnvironmentVariables(configurationSettingsEnvVariables);
+  }
+
+  @VisibleForTesting
+  public JTable getTable() {
+    return table;
+  }
+
+  /** Adds the Cloud Library environment variables to the selected App Engine run configurations. */
+  private class ApplyAction extends DialogWrapperAction {
+    private ApplyAction() {
+      super(CommonBundle.getApplyButtonText());
+    }
+
+    @Override
+    protected void doAction(ActionEvent event) {
+      addEnvironmentVariablesToConfiguration(getSelectedConfigurations());
+    }
+  }
+
+  /** The custom {@link JBTable} for the table of existing Google App Engine run configurations. */
+  private static final class ServerTable extends JBTable {
+
+    ServerTable(BooleanTableModel tableModel) {
+      super(tableModel);
+      setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+      setDefaultRenderer(
+          RunnerAndConfigurationSettings.class, new RunnerAndConfigurationSettingsRenderer());
+      setDefaultRenderer(Boolean.class, new BooleanTableCellRenderer());
+      setDefaultEditor(Boolean.class, new BooleanTableCellEditor());
+      TableUtil.setupCheckboxColumn(getColumnModel().getColumn(tableModel.getBooleanColumn()));
```

Comment:
This looks like a deprecated method, they suggest using `setupCheckboxColumn(JTable, colndex)` instead